common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -46,18 +46,22 @@
   private static final boolean unaligned;
   static {
     boolean _unaligned;
-    // use reflection to access unaligned field
-    try {
-      Class<?> bitsClass =
-        Class.forName("java.nio.Bits", false, ClassLoader.getSystemClassLoader());
-      Method unalignedMethod = bitsClass.getDeclaredMethod("unaligned");
-      unalignedMethod.setAccessible(true);
-      _unaligned = Boolean.TRUE.equals(unalignedMethod.invoke(null));
-    } catch (Throwable t) {
-      // We at least know x86 and x64 support unaligned access.
-      String arch = System.getProperty("os.arch", "");
-      //noinspection DynamicRegexReplaceableByCompiledPattern
-      _unaligned = arch.matches("^(i[3-6]86|x86(_64)?|x64|amd64|aarch64)$");
+    if (arch.matches("^(ppc64le | ppc64)$")) {
+      _unaligned = true;
+    } else {
+      //Since java.nio.Bits.unaligned() doesn't return true on ppc (See JDK-8165231)
```

Comment:
Moving this comment to line 50?

Maybe you also need to explain it more clearly.  
> Since java.nio.Bits.unaligned() doesn't return true on ppc (See JDK-8165231), but ppc64 and ppc64le support it